### PR TITLE
Fix clang tidy warnings

### PR DESCRIPTION
**Pull request by MiKom**
This is a giant PR but it's comprised of (mostly) small atomic commits. So best reviewed commit by commit.

Not _everything_ is fixed yet. Few more complicated ones are left out. Also, fixing warnings in tests will be done separately.

### CODE REVIEW

File: src/KDUtils/file_mapper.cpp [hunk 1]
```diff
@@ -21,9 +21,9 @@ using WriteMap = mio::basic_mmap<mio::access_mode::write, uint8_t>;
 struct FileMapper::Map {
     std::variant<ReadMap, WriteMap> map;
 
-    inline ~Map() {};
+    ~Map() {};
```
Comment (mkrus) [inline src/KDUtils/file_mapper.cpp:24]: remove trailing semi column?

File: src/KDGui/platform/linux/xcb/linux_xcb_platform_window.cpp [hunk 2]
```diff
@@ -41,8 +41,9 @@ bool LinuxXcbPlatformWindow::create()
     const auto screen = m_platformIntegration->screen();
 
     m_xcbWindow = xcb_generate_id(connection);
-    uint32_t mask = XCB_CW_EVENT_MASK;
-    uint32_t values[1] = {
+    const uint32_t mask = XCB_CW_EVENT_MASK;
+    // NOLINTNEXTLINE(modernize-avoid-c-arrays)
```
Comment (mkrus) [inline src/KDGui/platform/linux/xcb/linux_xcb_platform_window.cpp:45]: would it make sense to use std::array instead?

File: src/KDGui/platform/linux/wayland/linux_wayland_platform_window.cpp [hunk 3]
```diff
@@ -228,15 +230,15 @@ void LinuxWaylandPlatformWindow::handleTextInput(const std::string &str)
     CoreApplication::instance()->sendEvent(m_window, &ev);
 }
 
-void LinuxWaylandPlatformWindow::enter(wl_surface *surface, wl_output *output)
+void LinuxWaylandPlatformWindow::enter(wl_surface * /*surface*/, wl_output *output)
```
Comment (mkrus) [inline src/KDGui/platform/linux/wayland/linux_wayland_platform_window.cpp:233]: why not use KD_UNUSED?

File: src/KDGui/platform/linux/wayland/linux_wayland_platform_output.cpp [hunk 4]
```diff
@@ -59,7 +62,7 @@ void LinuxWaylandPlatformOutput::done(wl_output *output)
 {
 }
 
-void LinuxWaylandPlatformOutput::scale(wl_output *output, int32_t factor)
+void LinuxWaylandPlatformOutput::scale(wl_output * /*output*/, int32_t factor)
```
Comment (mkrus) [inline src/KDGui/platform/linux/wayland/linux_wayland_platform_output.cpp:65]: why not use KD_UNUSED?

File: src/KDGui/platform/linux/wayland/linux_wayland_platform_integration.cpp [hunk 5]
```diff
@@ -168,7 +170,7 @@ void LinuxWaylandPlatformIntegration::global(wl_registry *registry, uint32_t id,
     }
 }
 
-void LinuxWaylandPlatformIntegration::globalRemove(wl_registry *registry, uint32_t id)
+void LinuxWaylandPlatformIntegration::globalRemove(wl_registry * /*registry*/, uint32_t id)
```
Comment (mkrus) [inline src/KDGui/platform/linux/wayland/linux_wayland_platform_integration.cpp:173]: why not use KD_UNUSED?

File: src/KDGui/platform/linux/wayland/linux_wayland_platform_input.cpp [hunk 6]
```diff
@@ -79,7 +84,7 @@ void LinuxWaylandPlatformInput::destroyPointerConstraintsV1()
     }
 }
 
-void LinuxWaylandPlatformInput::capabilities(wl_seat *seat, uint32_t caps)
+void LinuxWaylandPlatformInput::capabilities(wl_seat * /*seat*/, uint32_t caps)
```
Comment (mkrus) [inline src/KDGui/platform/linux/wayland/linux_wayland_platform_input.cpp:87]: why not use KD_UNUSED? (and loads below)

**Review by mkrus — COMMENTED** [review]
wonder if exception handling should be a configuration option?

File: src/KDGui/platform/linux/wayland/linux_wayland_platform_window.cpp (hunk 3 above)
Comment (MiKom) [reply src/KDGui/platform/linux/wayland/linux_wayland_platform_window.cpp:233]: Not really a strong preference. I just thought that this way we won't introduce additional lines so just a little less visual clutter.

Comment (MiKom) [reply src/KDGui/platform/linux/wayland/linux_wayland_platform_window.cpp:233]: Done

File: src/KDUtils/file_mapper.cpp (hunk 1 above)
Comment (MiKom) [reply src/KDUtils/file_mapper.cpp]: Done

File: src/KDGui/platform/linux/xcb/linux_xcb_platform_window.cpp (hunk 2 above)
Comment (MiKom) [reply src/KDGui/platform/linux/xcb/linux_xcb_platform_window.cpp]: Done

**Comment by MiKom** [comment]
> wonder if exception handling should be a configuration option?

We throw exceptions in multiple places, particularly in initializing Wayland and XCB. Plus, KDBindings may throw as well and doesn't have any support for turning exceptions on and off. I suggest we create an issue for it and start working on it once the concrete need arises. Whad do you think @mkrus ?

**Review by mkrus — APPROVED** [review]
don't mind either way regarding KD_UNUSED vs commented out names, just saw KD_UNUSED used (sic) in some places

**Pull request merged**